app/components/UserValidator/index.js

```diff
@@ -26,15 +26,18 @@ const Validator = (props: Props) => {
   const input = useRef<?HTMLInputElement>(null);
   const [completed, setCompleted] = useState(false);
   const [showScanner, setShowScanner] = useState(false);
-  const [processing, setProcessing] = useState(false);
+  let processing = false;
 
   const showCompleted = () => {
     setCompleted(true);
-    setTimeout(() => setCompleted(false), 2000);
+    setTimeout(() => {
+      processing = false;
```

Comment:
Why not keep the `useState` and add run the `setProccessing` here? This seems very weird

Reply:
Because the state does not update quickly enough. When having a state variable, the handleSelect function will be called multiple times, since if (!processing) does not refer to the updated `processing` variable when using state variables.

Reply:
This causes double and triple scans, where the user receives messages that the user is already present, even though the just scanned them 100ms ago.

Reply:
This is because useState is asynchronous. ✌🏻

Reply:
I don't see how this could work, won't `processing` always be false in the actual component rendering?

Reply:
You might want to use a [useRef](https://reactjs.org/docs/hooks-reference.html#useref) if you need it to update before rerendering.

Reply:
This solution actually worked, @eikhr, probably because the component did not re render when processing was set? (Dunno)

However, I solved this now with hooks using a useEffect to detect when the scanned result changes, and then run the handleSelect. ✅